client/my-sites/plans/jetpack-plans/product-store/featured-item-card/index.tsx

```diff
@@ -16,10 +17,14 @@ export const FeaturedItemCard: React.FC< FeaturedItemCardProps > = ( {
 	onClickMore,
 	onClickPurchase,
 	siteId,
+	purchase,
 } ) => {
 	const translate = useTranslate();
 	const { displayName: title, description } = item;
 
+	const { isNotPlanOwner } = useItemInfo( { purchase } );
+	const shouldDisableButton = isOwned && isNotPlanOwner;
+
```

Comment:
Let us try to keep `FeaturedItemCard` clean and avoid any computational logic here. That way we can re-use it elsewhere.

So, we can pass a prop like `isCtaDisabled` and pass it from the parent/consumer component.